packages/keyring-controller/src/KeyringController.ts

```diff
@@ -2556,17 +2538,18 @@ export class KeyringController extends BaseController<
     this.#assertControllerMutexIsLocked();
 
     try {
-      const { type, data } = serialized;
+      const { type, data, metadata: serializedMetadata } = serialized;
       const keyring = await this.#createKeyring(type, data);
       // If metadata is missing, assume the data is from an installation before
       // we had keyring metadata.
-      if (this.#keyringsMetadata.length <= this.#keyrings.length) {
-        console.log(`Adding missing metadata for '${type}' keyring`);
-        this.#keyringsMetadata.push(getDefaultKeyringMetadata());
-      }
+      // WARN: This won't be persisted in the vault, so it will be lost on reload.
```

Comment:
I guess you mean it won't get persisted until the next vault update? So it _might_ be lost on reload? Not definitely lost

Reply:
yeah exactly, I placed this here so I don't forget to add a vault update on login somewhere before making the PR ready. The main thing I'm trying to figure out is how to detect that new metadata has been generated (so that we don't have to update the vault after every login)

Reply:
Done in https://github.com/MetaMask/core/pull/5725/commits/f753328ff05c6a1a2fb9766c358b5f5e202142f6